src/lib/mongoBackend/mongoSubCache.cpp

```diff
@@ -710,6 +744,11 @@ void mongoSubCountersUpdate
     mongoSubCountersUpdateCount(db, COL_CSUBS, subId, count);
   }
 
+  if (failsCounter > 0)
+  {
+    mongoSubCountersUpdatefailsCounter(db, COL_CSUBS, subId, failsCounter);
```

Comment:
Maybe instead of defining a new function for this case (mongoSubCountersUpdatefailsCounter) maybe you can extend the mongoSubCountersUpdateCount() function, calling it this way (line 744):

```
mongoSubCountersUpdateCount(db, COL_CSUBS, subId, count, failsCounter);
```

and do the `failsCounter > 0` inside mongoSubCountersUpdateCount()

Reply:
Fixed in https://github.com/telefonicaid/fiware-orion/pull/3890/commits/0d976f5d55b682e27c153ecd5662ef408f351a1e